commands/scale/cluster/command.go

```diff
@@ -174,16 +174,9 @@ func getClusterStatus(clusterID, activityName string) (*client.ClusterStatus, er
 	auxParams.ActivityName = activityName
 
 	// Make sure we have provider info in the current endpoint
-	if config.Config.Provider == "" {
-		if flags.CmdVerbose {
-			fmt.Println(color.WhiteString("Fetching provider information"))
-		}
-
-		info, err := clientV2.GetInfo(auxParams)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-		config.Config.SetProvider(info.Payload.General.Provider)
+	err = clientV2.EnsureProviderInConfig()
```

Comment:
When we do it this way we need to add it to all the commands. When we want to do it like this I would suggest a different entrypoint. Doing it for the "scale cluster" command in `getClusterStatus` is a snowflake and not straight forward. For other commands it would then look differently, which already smells. 

When we want to do it for every command separately, I would suggest a more straight forward entrypoint. But then it should always be the same. This would require a fair amount of refactoring I think. 

When you want to just make it work I thought we talked about the client initialization `NewWithConfig`. There we could then call `ensureProviderInConfig` before returning the client. It would then not be a so super cool design, but the most easy solution for now without a lot of refactoring.

Reply:
All right. Will change.

Reply:
Changed. PTAL